docs/proposals/20221611-workload-identity-integration.md

```diff
@@ -0,0 +1,374 @@
+
+The service account that the capz pod will use should be annotated with the client ID of the AAD application or User Assigned Identity and a federated identity should be created by associating the Service Account as a subject.
+
+```yaml
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  annotations:
+    azure.workload.identity/client-id: ${APPLICATION_CLIENT_ID:-$USER_ASSIGNED_IDENTITY_CLIENT_ID}
+  labels:
+    azure.workload.identity/use: "true"
+  name: ${SERVICE_ACCOUNT_NAME}
+  namespace: ${SERVICE_ACCOUNT_NAMESPACE}
+```
+
+Also, a federated identity should be created using the azure cli ( or via the azure portal).
+
+```bash
+az identity federated-credential create \
+  --name "kubernetes-federated-credential" \
+  --identity-name "${USER_ASSIGNED_IDENTITY_NAME}" \
+  --resource-group "${RESOURCE_GROUP}" \
+  --issuer "${SERVICE_ACCOUNT_ISSUER}" \
+  --subject "system:serviceaccount:${SERVICE_ACCOUNT_NAMESPACE}:${SERVICE_ACCOUNT_NAME}"
+```
+
+#### <a name='DistributeKeys'></a>Distribute Keys To Management Cluster
+
+Setting up credentials for management cluster which is created from the bootstrap cluster is a key activity. This requires storing the keys on control plane node and patching kube-apiserver and kube-controller-manager flags to include the service account signing flags similar to what is discussed in the above `Set Service Account Signing Flags` section.
+
+### <a name='ProposedApiChanges'></a>Proposed API Changes
+
+The AzureClusterIdentity spec has a `Type` field that can be used to define what type of Azure identity should be used.
+
+```go
+// AzureClusterIdentitySpec defines the parameters that are used to create an AzureIdentity.
+type AzureClusterIdentitySpec struct {
+	// Type is the type of Azure Identity used.
+	// ServicePrincipal, ServicePrincipalCertificate, UserAssignedMSI or ManualServicePrincipal.
+	Type IdentityType `json:"type"`
+
+	// ...
+	// ...
+}
+```
+
+Introducing one more acceptable value for `Type` in AzureClusterIdentity spec for workload identity is proposed.
+
+```go
+// IdentityType represents different types of identities.
+// +kubebuilder:validation:Enum=ServicePrincipal;UserAssignedMSI;ManualServicePrincipal;ServicePrincipalCertificate;WorkloadIdentity
+type IdentityType string
+
+const (
+	// UserAssignedMSI represents a user-assigned managed identity.
+	UserAssignedMSI IdentityType = "UserAssignedMSI"
+
+	// ServicePrincipal represents a service principal using a client password as secret.
+	ServicePrincipal IdentityType = "ServicePrincipal"
+
+	// ManualServicePrincipal represents a manual service principal.
+	ManualServicePrincipal IdentityType = "ManualServicePrincipal"
+
+	// ServicePrincipalCertificate represents a service principal using a certificate as secret.
+	ServicePrincipalCertificate IdentityType = "ServicePrincipalCertificate"
+	
+	//[Proposed Change] WorkloadIdentity represents  azure workload identity.
+	WorkloadIdentity IdentityType = "WorkloadIdentity"
```

Comment:
Q: Are those fields immutable?
Just wondering what happens if the user changes settings in this struct on an existing cluster

slightly related

Q: What is the migration path for users currently using the deprecate azure pod identity?